packages-content-model/roosterjs-content-model-core/lib/corePlugin/ContentModelCopyPastePlugin.ts

```diff
@@ -250,6 +256,45 @@ class ContentModelCopyPastePlugin implements PluginWithState<CopyPastePluginStat
     };
 }
 
+/**
+ * @internal
+ * Exported only for unit testing
+ */
+export function adjustSelectionForCopyCut(pasteModel: ContentModelDocument) {
+    let selectionMarker: ContentModelSelectionMarker | undefined;
+    let firstBlock: ContentModelBlock | undefined;
+
+    iterateSelections(pasteModel, (_, tableContext, block, segments) => {
+        if (selectionMarker) {
+            if (
+                tableContext &&
+                block &&
+                firstBlock?.blockType == 'Paragraph' &&
+                firstBlock.segments.includes(selectionMarker) &&
+                block.blockType == 'Paragraph'
+            ) {
+                block.segments?.unshift(selectionMarker);
+                firstBlock.segments.splice(firstBlock.segments.indexOf(selectionMarker), 1);
+            }
+            return true;
+        }
+
+        if (
+            !selectionMarker &&
+            segments?.length == 1 &&
+            segments[0].segmentType == 'SelectionMarker'
+        ) {
+            if (tableContext) {
```

Comment:
do we need this check?

What if there is a table nested in another table, the select content from the first cell of inner table?

Reply:
Good point, removing this to handle this case.